src/Evolution/Systems/GeneralizedHarmonic/Equations.hpp

```diff
@@ -127,4 +142,147 @@ struct ComputeNormalDotFluxes {
+ *
+ * The upwind flux in general is given by Eq. (6.3) of \cite Teukolsky2015ega :
+ * \f{eqnarray}{
+ * D = S \Lambda^{\rm pos} S^{-1} u^{\rm int}
+ *   + S \Lambda^{\rm neg} S^{-1} u^{\rm ext},
+ * \f}
+ * where \f$u\f$ is a vector of the evolved variables, \f$S^{-1}\f$
+ * maps the evolved variables into the characteristic variables \f$S^{-1}u\f$,
+ * \f$S\f$ maps the characteristic variables into the evolved variables,
+ * and \f$\Lambda\f$ is a diagonal matrix of the average characteristic
+ * speed at the interface.
+ *
+ * Here, \f$S^{-1}u^{\rm int}\f$ represents the
+ * characteristic variables at the element interface, computed using
+ * evolved variables from the element interior;
+ * \f$S^{-1}u^{\rm ext}\f$ represents the characteristic variables at the
+ * element interface, computed using evolved variables from the exterior,
+ * neighboring element; \f$\Lambda^{\rm pos}\f$ is a diagonal
+ * matrix whose nonzero entries are the average characteristic speeds
+ * that are positive ("outgoing", i.e. leaving the element); and
+ * \f$\Lambda^{\rm neg}\f$ is a diagonal matrix whose nonzero entries are
+ * the average characteristic speeds that are negative ("incoming", i.e.
+ * entering the element). If a characteristic field
+ * \f$U^{\rm int}\f$ has a characteristic speed \f$v^{\rm int}\f$ and the same
+ * field in the exterior \f$U^{\rm ext}\f$ has speed \f$v^{\rm ext}\f$, then
+ * the average characteristic speed is
+ * \f$v^{\rm avg} = (1/2)(v^{\rm int} + v^{\rm ext})\f$.
+ *
+ * This function implements the upwind flux for the generalized harmonic
+ * system. First, it computes the characteristic variables using i) the
+ * evolved variables from the interior and ii) the evolved variables from the
+ * exterior. Then, it computes \f$\Lambda^{\rm pos}\f$ and
+ * \f$\Lambda^{\rm neg}\f$ from the average characteristic speeds.
+ * Then, it computes the combination
+ * \f$\Lambda^{\rm pos} S^{-1} u^{\rm int}
+ * + \Lambda^{\rm neg} S^{-1} u^{\rm ext}\f$. Finally, it
+ * applies \f$S\f$ by converting the result back from characteristic to
+ * evolved variables, using the unit normal vector of the element and
+ * the average value of the field
+ * \f$\gamma_2^{\rm avg} = (1/2)(\gamma_2^{\rm int} + \gamma_2^{\rm ext})\f$,
+ * where here \f$\gamma_2^{\rm int}\f$ is the value of \f$\gamma_2\f$
+ * in the interior and \f$\gamma_2^{\rm ext}\f$ is the value of
+ * \f$\gamma_2\f$ in the exterior.
+ *
+ * \warning as currently implemented, this flux does not satisfy the
+ * generalized Rankine-Hugoniot condition
+ * \f$D^{\rm int}([S^{-1} u]^{\rm int}, [S^{-1} u]^{\rm ext})
+ * = - D^{\rm ext}([S^{-1} u]^{\rm ext}, [S^{-1} u]^{\rm int})\f$,
+ * which enforces that the flux leaving the interior is the flux entering
+ * the exterior and vice versa. This condition is necessary for a
+ * well-balanced scheme. Use this flux with caution; so far it has
+ * been tested for evolving Kerr-Schild in generalized harmonic.
```

Comment:
in `generalized harmonic *gauge*`

Reply:
No, in the generalized harmonic formulation of the Einstein equations.

Reply:
I updated this line anyway, since now I've also tested for a perturbed black hole. I no longer mention generalized harmonic at all, for simplicity. (GH is the only formulation we have implemented for GR, and the only one we will implement any time soon).